applications/TopologyOptimizationApplication/custom_elements/small_displacement_simp_element.cpp

```diff
@@ -0,0 +1,302 @@
+
+SmallDisplacementSIMPElement::SmallDisplacementSIMPElement( IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties )
+        : BaseType( NewId, pGeometry, pProperties )
+{
+    //DO NOT ADD DOFS HERE!!!
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Create( IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties ) const 
+{
+	return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Create( IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties ) const
+{
+	return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, pGeom, pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+SmallDisplacementSIMPElement::~SmallDisplacementSIMPElement()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Clone (
+    IndexType NewId,
+    NodesArrayType const& rThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    SmallDisplacementSIMPElement::Pointer p_new_elem = Kratos::make_intrusive<SmallDisplacementSIMPElement>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_elem->SetData(this->GetData());
+    p_new_elem->Set(Flags(*this));
+
+    // Currently selected integration methods
+    p_new_elem->SetIntegrationMethod(BaseType::mThisIntegrationMethod);
+
+    // The vector containing the constitutive laws
+    p_new_elem->SetConstitutiveLawVector(BaseType::mConstitutiveLawVector);
+
+    return p_new_elem;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+
+// =============================================================================================================================================
+// STARTING / ENDING METHODS
+// =============================================================================================================================================
+
+//************************************************************************************
+//************************************************************************************
+
```

Comment:
I will review this after some clean up in the other files (but avoid to duplicate code already defined in the StructuralMechanics app)